lib/atomic-gssize.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2002-2018 Balabit
+ * Copyright (c) 2018 Laszlo Budai <laszlo.budai@balabit.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef ATOMIC_GSSIZE_H_INCLUDED
+#define ATOMIC_GSSIZE_H_INCLUDED
+
+typedef struct
+{
+  gssize value;
+} atomic_gssize;
+
+static inline gssize
+atomic_gssize_add(atomic_gssize *a, gssize add)
+{
+  return g_atomic_pointer_add(&a->value, add);
+}
+
+static inline gssize
+atomic_gssize_sub(atomic_gssize *a, gssize sub)
+{
+  return g_atomic_pointer_add(&a->value, -1 * sub);
+}
+
+static inline gssize
+atomic_gssize_inc(atomic_gssize *a)
+{
+  return g_atomic_pointer_add(&a->value, 1);
+}
+
+static inline gssize
+atomic_gssize_dec(atomic_gssize *a)
+{
+  return g_atomic_pointer_add(&a->value, -1);
+}
+
+static inline gssize
+atomic_gssize_get(atomic_gssize *a)
+{
+  return (gssize)g_atomic_pointer_get(&a->value);
+}
+
+static inline void
+atomic_gssize_set(atomic_gssize *a, gssize value)
+{
+  g_atomic_pointer_set(&a->value, value);
+}
+
+static inline gsize
+atomic_gssize_get_unsigned(atomic_gssize *a)
+{
+  return (gsize)g_atomic_pointer_get(&a->value);
+}
+
+static inline gssize
+atomic_gssize_racy_get(atomic_gssize *a)
```

Comment:
I'd remove these 3 racy operations. We had only problems with them in the past.

Reply:
...and I would not, not part of this PR. I agree with you that we should consider to remove all the racy atomic operations, but... but I also know that I don't want to discuss it it as part of this PR.
This is why I wanted to atomic-gssize PR handled separately. Please, add these notes to there, or, after that PR merged, we could open a new PR where we remove all these operations and explain why we don't need them anymore.

If no other issues, don't forget to approve the requested changes :)

Reply:
Okay. I've added my comment to #2159.